lib/adbc_connection.ex

```diff
@@ -85,6 +172,30 @@ defmodule Adbc.Connection do
     end
   end
 
+  @doc """
+  Runs the given `query` with `params`.
+  """
+  @spec query_with_options(t(), binary | reference, [term], Keyword.t()) ::
+          {:ok, result_set} | {:error, Exception.t()}
+  def query_with_options(conn, query, params \\ [], statement_options)
```

Comment:
Instead of introducing `query_with_options` why not rely on `query/4` instead? Pretty much we can rename this function to `query` and remove the existing `query/3`, WDYT?

Reply:
Acutally yes, that's a good idea. I'll update it now/